pkg/tfgen/installation_docs.go

```diff
@@ -124,3 +144,192 @@ func applyEditRules(contentBytes []byte, docFile *DocFile) ([]byte, error) {
 	}
 	return contentBytes, nil
 }
+
+func translateCodeBlocks(contentStr string, g *Generator) (string, error) {
+
+	var returnContent string
+
+	examples := map[string]string{}
+	// Extract code blocks
+	codeFence := "```"
+	var codeBlocks []codeBlock
+	for i := 0; i < (len(contentStr) - len(codeFence)); i++ {
+		block, found := findCodeBlock(contentStr, i)
+		if found {
+			codeBlocks = append(codeBlocks, block)
+			i = block.end + 1
+		}
+	}
+	if len(codeBlocks) == 0 {
+		return contentStr, nil
+	}
+	startIndex := 0
+	outDir, err := os.MkdirTemp("", "configuration-example")
+	if err != nil {
+		return "", err
+	}
+	// In order to translate the examples with the currently-being-generated version of our provider, we will obtain
+	// mappings from the cli converter.
+	mappingsDir := filepath.Join(outDir, "mappings")
+	mappings := []tfbridge.ProviderInfo{
+		g.cliConverter().info,
+	}
+	// Prepare mappings folder
+	if len(mappings) > 0 {
+		if err := os.MkdirAll(mappingsDir, 0755); err != nil {
+			return "", fmt.Errorf("translateCodeBlocks: failed to write mappings folder: %w", err)
+		}
+	}
+	// Write out mappings files
+	for _, info := range mappings {
+		info := info
+		marshallProviderInfo := tfbridge.MarshalProviderInfo(&info)
+		bytes, err := json.Marshal(marshallProviderInfo)
+		if err != nil {
+			return "", fmt.Errorf("translateCodeBlocks: failed to write mappings folder: %w", err)
+		}
+		mappingsFile := g.cliConverter().mappingsFile(mappingsDir, info)
+		if err := os.WriteFile(mappingsFile, bytes, 0600); err != nil {
+			return "", fmt.Errorf("translateCodeBlocks: failed to write mappings file: %w", err)
+		}
+	}
+	var mappingsArgs []string
+	for _, info := range mappings {
+		mappingsArgs = append(mappingsArgs, "--mappings", g.cliConverter().mappingsFile(mappingsDir, info))
+	}
+	var cleanupErr error
+	defer func() {
+		if err := os.RemoveAll(outDir); err != nil {
+			cleanupErr = fmt.Errorf("failed to clean up configuration-example dir: %w", err)
+		}
+	}()
+	for i, block := range codeBlocks {
+		// Write the content up to the start of the code block
+		returnContent = returnContent + contentStr[startIndex:block.start]
+		nextNewLine := strings.Index(contentStr[block.start:block.end], "\n")
+		if nextNewLine == -1 {
+			//Write the inline block
+			returnContent = returnContent + contentStr[block.start:block.end] + codeFence + "\n"
+			continue
+		}
+		fenceLanguage := contentStr[block.start : block.start+nextNewLine+1]
+		code := contentStr[block.start+nextNewLine+1 : block.end]
+		// Only convert code blocks that we have reasonable suspicion of actually being Terraform.
+		if fenceLanguage == "```terraform\n" || fenceLanguage == "```hcl\n" ||
+			(fenceLanguage == "```\n" && guessIsHCL(code)) {
+
+			//Generate the main.tf file for converting the config file, since --convert-examples doesn't support
+			// the Pulumi.yaml config file
+			err := os.WriteFile(filepath.Join(outDir, "main.tf"), []byte(code), 0600)
```

Comment:
Should this be a `tmp` file? If not, should there be a `defer` block to remove it? I don't think we want to leave a `.tf` file in our output tree.